### DISCUSSION TITLE
GitHub pagees works just for one site

### DISCUSSION THREAD

**Discussion by stasmotorny** [community/community#22108] [Other Feature Feedback, Questions, & Ideas]
<p>Hi. I have repository <a href="https://github.com/stasmotorny/stasmotorny.github.io" rel="nofollow noopener">https://github.com/stasmotorny/stasmotorny.github.io</a> I set a custom domain frontdev.pp.ua for this repository.<br>
This site works fine, but I can’t run all other my sites. Instead of them I see a start page of  frontdev.pp.ua for example you can try this link <a href="https://stasmotorny.github.io/photographer/" rel="nofollow noopener">https://stasmotorny.github.io/photographer/</a><br>
Here is my domain dns settigs.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/7/767d32e313239327d928f3c9e9b294c2dea29a9f.png" data-download-href="/uploads/short-url/gUczSm207cdDsHTmD2wmutKXRwz.png?dl=1" title="dns settings" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/33233429/181068904-d8be20bd-1ab3-4098-84bd-b062b624a228.png" alt="dns settings" data-base62-sha1="gUczSm207cdDsHTmD2wmutKXRwz" width="690" height="226" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/7/767d32e313239327d928f3c9e9b294c2dea29a9f_2_690x226.png, https://aws1.discourse-cdn.com/github/original/2X/7/767d32e313239327d928f3c9e9b294c2dea29a9f.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/7/767d32e313239327d928f3c9e9b294c2dea29a9f.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/7/767d32e313239327d928f3c9e9b294c2dea29a9f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">dns settings</span><span class="informations">806×264 11.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Is it possible to fix this problem?</p>

**Comment by otter-computer** [comment, marked as answer]
<p>This is happening because you’ve got a service worker installed for the site hosted from your <code>stasmotorny/stasmotorny.github.io</code> repository.</p>
<p>When you add a custom domain to a repository using the <code>*.github.io</code> naming scheme this custom domain applies to <em>all</em> sites hosted from your account.</p>
<p>When the service worker is installed it will capture all traffic to sites hosted from the same domain <code>frontdev.pp.ua</code> and redirect it to the root index page of your site. As all Pages sites from your account are now hosted from this domain all visits to your other site are being redirected to the site hosted from <code>stasmotorny/stasmotorny.github.io</code>.</p>
<p>There are 2 options for how to fix this:</p>
<ol>
<li>
<p>Disable the service worker for the site hosted in <code>stasmotorny/stasmotorny.github.io</code>. You’ll need to look up how to do this in the React documentation and commit that change to your repository.</p>
</li>
<li>
<p>Rename your <code>stasmotorny/stasmotorny.github.io</code> repository to something other than the <code>*.github.io</code> naming scheme to make your other sites not inherit this domain. This will stop visits to your other site from redirecting to the root index, but it will mean the URLs of these sites will be changed from <code>frontdev.pp.ua/repository-name</code>.</p>
</li>
</ol>
